brian2/core/variables.py

```diff
@@ -274,6 +275,9 @@ def __repr__(self):
                                   constant=repr(self.constant),
                                   read_only=repr(self.read_only))
 
+    _state_tuple = property(lambda self: (self.dim, self.dtype, self.scalar,
+                                          self.constant, self.read_only,
+                                          self.dynamic, self.name))
```

Comment:
I don't think we can automatize the `_state_tuple` here: it's very point is to make the distinction between attributes that matter for cacheing and those that don't. E.g., for arrays all describing attributes (`dtype`, `dim`, etc.) matter, but `value` does not. For a `Constant`, on the other hand, the `value` matters (because it might be used with its value in the generated code).